src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8770,7 +8770,11 @@ internal static void DeleteFileStream(string path, string streamName)
             }
             string resultPath = path + adjustedStreamName;
 
-            NativeMethods.DeleteFile(resultPath);
+            if (!NativeMethods.DeleteFile(resultPath))
+            {
+                int error = Marshal.GetLastWin32Error();
+                throw new Win32Exception(error);
```

Comment:
Good catch! 
`AlternateDataStreamUtilities` uses the pattern. So maybe add `try-catch` in Unblock-File cmdlet file?